### PR TITLE
[SYCLomatic] Support coloring the output of query-api-mapping.

### PR DESCRIPTION
**Pull request by tangjj11**
Signed-off-by: Tang, Jiajun jiajun.tang@intel.com

### CODE REVIEW

File: clang/lib/DPCT/DPCT.cpp [hunk 1]
```diff
@@ -808,7 +808,9 @@ int runDPCT(int argc, const char **argv) {
     }
     StartPos = StartPos + StartStr.length();
     EndPos = SourceCode.find_last_of('\n', EndPos);
-    QueryAPIMappingStr += SourceCode.substr(StartPos, EndPos - StartPos + 1);
+    QueryAPIMappingStr +=
+        "\033[32m" + SourceCode.substr(StartPos, EndPos - StartPos + 1).str() +
```
Comment (zhimingwang36) [inline clang/lib/DPCT/DPCT.cpp]: suggest to define a Macor/constExpr to replace the "033[32m" with a right name to improve readability.

File: clang/lib/DPCT/DPCT.cpp [hunk 2]
```diff
@@ -1130,6 +1132,7 @@ int runDPCT(int argc, const char **argv) {
     static const std::string StartStr{"// Start"};
     static const std::string EndStr{"// End"};
     bool Flag = false;
+    llvm::outs() << llvm::raw_ostream::BLUE;
```
Comment (zhimingwang36) [inline clang/lib/DPCT/DPCT.cpp:1135]: if the output color is controlled by here, then don't try to embed raw string with color control

Comment (tangjj11) [reply clang/lib/DPCT/DPCT.cpp:1135]: fixed

File: clang/lib/DPCT/DPCT.cpp (hunk 1 above)
Comment (tangjj11) [reply clang/lib/DPCT/DPCT.cpp]: fixed the next comment and this problem is not exist.

**Review by wenhuiNi — APPROVED** [review]

**Review by yihwang-nv — APPROVED** [review]

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**